templates/huginn/0/README.md

```diff
@@ -0,0 +1,25 @@
+# Huginn
+
+Your agents are standing by!
+
+[Huginn](https://github.com/huginn/huginn) is a tool that allows you to automate tasks on the internet and/or your private intranet.
+
+## Installation
+
+When huginn first starts up it will create one user named, `admin`, with the password, `password`, along with a number of example agents for that user. Each new user will start off with a set of example agents.
+
+**Warning**: The **Require Confirmed Email** option *must* be **off** when you first start up the stack because the default user has a fake email address and will not allow you to login. If desired, you can upgrade the stack and turn email verification back on after changing the admin's email address.
+
+The Huggin web interface is served in the `huginn_web` container on port `3000`. In order to get web traffic to the container you have to create a Rancher load balancer and route the traffic to the container on port `3000`.
```

Comment:
I would add load balancer service to publish the service.

Reply:
The problem I have with having a Rancher Template publish *any* ports on the host, is that there is an issue if you want to have multiple applications running on that host port. Allowing you to pick which port you want to publish does do a little good, but you still have problems if you want to host multiple apps on the same port.

For example if I have only one Rancher host, and I want to serve App1 and App2 on port 80, I can't use the Rancher templates for App1 and App2 because they both need to bind port 80 on the host that they are running on. Even if I have more than one host, I still have to coordinate which apps run on which host and the limit to the number of apps I can serve is the number of hosts that are in the cluster.

By creating a global scale Rancher Load Balancer that binds 80 on all hosts and forwards the traffic to the correct service based on incoming domain, I can serve any number of web applications on any number of hosts without any port conflicts. Yes, I have to manually create the load balancer rules, but it's not really that bad and it lets me serve my apps without having to worry about port conflicts.

Optimally we would be able to put a question in the rancher-compose that would ask if you wanted the template to deploy a load balancer, but as far as I am aware there isn't a way to do that with docker-compose templates.

In the end, it is a little bit more work to create the load balancer yourself instead of having the template deploy one, but it is more flexible. I simply would not be able to deploy the Huginn template on my Rancher cluster if it has to bind port 80 or 443. Yes, if it gave me the choice of which port to bind, I could have it bind some random port I don't care about, but then I am opening a port on my host for no reason other than that it was what the template required me to do.

I definitely understand wanting the template to deploy the load balancer for you, but I think the fact that it rules out a use case entirely is a good reason not to do it. If in light of these concerns you still think that we should put the load balancer in the template, I will comply because it *is* Rancher's community catalog, I will just have to maintain my own version of the template in a private catalog.

Reply:
I get your point. It's not mandatory just a proposal to get better package. :)